api/dataset.go

```diff
@@ -136,29 +136,57 @@ func (api *DatasetAPI) getEditions(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	bytes, err := json.Marshal(results)
-	if err != nil {
-		log.ErrorC("failed to marshal a list of edition resources into bytes", err, log.Data{"dataset_id": id})
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
+	// If auth, we only return the .current document of editions (if a current doc exists then it's state is always published).
+	if auth == true {
+		bytes, err := json.Marshal(results)
+		if err != nil {
+			log.ErrorC("failed to marshal a list of edition resources into bytes", err, log.Data{"dataset_id": id})
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		setJSONContentType(w)
+		_, err = w.Write(bytes)
+		if err != nil {
+			log.Error(err, log.Data{"dataset_id": id})
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		log.Debug("get all editions with auth", log.Data{"dataset_id": id})
```

Comment:
From line 148 to 154 the code is repeated in the else clause (171 to 177). Could this be moved out the if/else clause?